plugin-locationcomponent/src/main/java/com/mapbox/maps/plugin/locationcomponent/LocationCompassEngine.kt

```diff
@@ -57,6 +59,25 @@ internal class LocationCompassEngine(context: Context) : SensorEventListener {
     }
   }
 
+  fun addCalibrationListener(compassCalibrationListener: LocationCompassCalibrationListener) {
+    calibrationListeners.add(compassCalibrationListener)
+  }
+
+  fun removeCalibrationListener(compassCalibrationListener: LocationCompassCalibrationListener) {
+    calibrationListeners.remove(compassCalibrationListener)
+  }
+
+  override fun onAccuracyChanged(sensor: Sensor?, accuracy: Int) {
+    if (accuracy == SensorManager.SENSOR_STATUS_UNRELIABLE) {
+      logD(TAG, "Compass sensor is unreliable, device calibration is needed.")
+      if (calibrationListeners.isNotEmpty()) {
```

Comment:
not sure that this check necessary here. if list is empty you will not iterate.

Reply:
this is a habit for when using CopyOnWriteArrayLists, in that case it would allocate an iterator but you are right this is not needed here.